db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryComparisonOperatorNode.java

```diff
@@ -495,4 +494,36 @@ public ValueNode genSQLJavaSQLTree() throws StandardException
 
 		return this;
 	}
+
+	@Override
+	public ValueNode replaceIndexExpression(ResultColumnList childRCL) throws StandardException {
+		if (childRCL == null) {
+			return this;
+		}
+		if (receiver != null) {
```

Comment:
There is no need to check `receiver`. It is either left operand or right operand and used at the code generation phase.

Reply:
Done.